src/nvim/message.c

```diff
@@ -3159,13 +3162,14 @@ void msg_ext_ui_flush(void)
   msg_ext_emit_chunk();
   if (msg_ext_chunks->size > 0) {
     Array *tofree = msg_ext_init_chunks();
-    ui_call_msg_show(cstr_as_string(msg_ext_kind), *tofree, msg_ext_overwrite);
+    ui_call_msg_show(cstr_as_string(msg_ext_kind), *tofree, msg_ext_overwrite, msg_ext_history);
```

Comment:
is it useful to pass the index? 
```suggestion
    ui_call_msg_show(cstr_as_string(msg_ext_kind), *tofree, msg_ext_overwrite,
                     msg_ext_history ? msg_hist_len : -1);
```

Reply:
Inclined to say no, even though I'm still not sure how useful the argument itself is. Just added it because of #31643, and because @bfredl said: "I just don't see any reason to not expose it. Like it is observable state anyway, this just gives the information at an earlier time."

But since the message history is a FIFO, there is no guarantee that the message will still be at the same index when the history is inspected, unless done immediately, at which point it will simply be the last element.